src/plugins/data/public/search/aggs/utils/prop_filter.ts

```diff
@@ -28,7 +28,7 @@ type FilterFunc<P extends keyof T, T> = (item: T[P]) => boolean;
  *
  * @returns the filter function which can be registered with angular
  */
-function propFilter<P extends string>(prop: P) {
+export function propFilter<P extends string>(prop: P) {
```

Comment:
`propFilter` had to stay behind for now as the `field` agg param type relies on it. All other filter-related items were moved to `vis_default_editor`